Automattic/wp-calypso
# Social: Landing page pricing tweaks.

This PR enables you to select multiple plans of Jetpack Social.  The variant selection for checking out the advanced plan isn't working currently on calypso blue correctly. It doesn't show the monthly variant when you go to the checkout page
#### Proposed Changes

* The changes would have been a lot simpler if we decided to do the same thing that backup does, but since we decided to not show the radio buttons, the product-lightbox changes needed to be a little bit more social specific, as opposed to being generic. The changes are mainly around the multiplechoice question component and adding some additional props to the PaymentPlan component.
* I have introduced the `jetpack-social/advanced-plan` feature flag, so the changes are picked up only on local dev environment.

How Social looks:

<img width="1226" alt="Screenshot 2023-01-13 at 18 49 09" src="https://user-images.githubusercontent.com/6594561/212329252-94f6c789-f789-4a33-bf19-5a2c8c907396.png">

How Backup looks:

<img width="1249" alt="Screenshot 2023-01-13 at 18 50 03" src="https://user-images.githubusercontent.com/6594561/212329382-98f7d1f8-3748-43e7-af4a-550b1b2f3ffe.png">

#### Testing Instructions



* Checkout to this branch and do `yarn start-jetpack-cloud` and go to http://jetpack.cloud.localhost:3000/pricing?view=products
*  Clicking on Social's GET button and more about Social button should open the modal.
* After opening the modal, select the plan of your choice and finish checkout.

To test plan's variant selections b/w the monthly and yearly plans for the advanced plan
*  Do `yarn start` to boot calypso
* Go to http://calypso.localhost:3000/checkout/jetpack/jetpack_social_advanced_yearly to make sure the monthly variant is shown as well. 
<img width="553" alt="Screenshot 2023-01-13 at 18 56 16" src="https://user-images.githubusercontent.com/6594561/212330552-5d27894e-2365-45a1-a9c0-d5ca80e4a9bb.png">

#### Pre-merge Checklist
